app/assets/javascripts/govuk_publishing_components/components/search-with-autocomplete.js

```diff
@@ -0,0 +1,99 @@
+/* global accessibleAutocomplete, fetch */
+//= require accessible-autocomplete/dist/accessible-autocomplete.min.js
+
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  class SearchWithAutocomplete {
+    constructor ($module) {
+      this.$module = $module
+
+      this.$originalInput = this.$module.querySelector('input')
+      this.$inputWrapper = this.$module.querySelector('.gem-c-search__input-wrapper')
+      this.$form = this.$module.closest('form')
+
+      this.source = this.$module.getAttribute('data-source')
+      this.sourceKey = this.$module.getAttribute('data-source-key')
+    }
+
+    init () {
+      // Do not enhance the input field if we won't be able to provide autocomplete suggestions
+      if (!this.source || !this.sourceKey) {
+        console.warn('search-with-autocomplete: No source/sourceKey provided')
+        return
+      }
+
+      const configOptions = {
+        element: this.$inputWrapper,
+        id: this.$originalInput.id,
+        name: this.$originalInput.name,
+        inputClasses: this.$originalInput.classList,
+        defaultValue: this.$originalInput.value,
+        confirmOnBlur: false,
+        showNoOptionsFound: false,
+        source: this.getResults.bind(this),
+        onConfirm: this.submitContainingForm.bind(this),
+        templates: {
+          suggestion: this.constructSuggestionHTMLString.bind(this)
+        }
+      }
+      accessibleAutocomplete(configOptions)
+
+      // The accessible-autocomplete component is meant to generate a new input element rather than enhancing an existing one, so we need to do some cleanup here.
+      const autocompleteInput = this.$inputWrapper.querySelector('.autocomplete__input')
+      // Ensure the new input element generated by accessible-autocomplete has the correct type
+      autocompleteInput.setAttribute('type', 'search')
+      // Remove the original input from the DOM
+      this.$originalInput.parentNode.removeChild(this.$originalInput)
+    }
+
+    // Callback used by accessible-autocomplete to generate the HTML for each suggestion based on
+    // the values returned from the source
+    constructSuggestionHTMLString (result) {
+      // We should be able to assume that data coming back from the Autocomplete API is safe to
+      // render as HTML. However, this is a final line of defence to prevent XSS.
+      const sanitizedResult = result.replace(/[&<>"']/g, '')
```

Comment:
Note: this is intentionally quite aggressive for now to err on the side of caution, at some point we can have a discussion about whether it's actually possible for a broken result to come back from the search engine.

Reply:
I don't know if I'm really just overthinking this. This deals with the edge case where Vertex AI Search (via Search API v2) returns search suggestions that contain something that looks like HTML.

We have no evidence of that happening in all our testing, the data is sanitised before it goes into VAIS to begin with, and it's vanishingly unlikely that anything nefarious could come through. If we do want to add a layer of sanitising, it would probably also be easier to do this on Search API v2 anyway!

Reply:
Happy to chat verbally if easier in case we're on different wavelengths